### DISCUSSION TITLE
希望短字符和长字符的翻译能进行区分

### DISCUSSION THREAD

**Discussion by jiazhao2048** [immersive-translate/immersive-translate#1286] [Ideas]
### 新功能 | New Feature

短字符翻译逻辑

### 为什么您想要添加这些功能 | Why you want to add these features

通过改写openai的prompt可以明显提升翻译质量。然而，改进的prompt对"短文本"例如单词的翻译token数量会过于“浪费”，而且有时候也prompt似乎对短文本容易识别错误，从而把prompt翻译出来，让短文本的翻译会出错。

留意到有“翻译段落所需的最少字符数”的功能，对于短文本则不进行翻译，一定程度上能够改善体验。而如果增加长短文本的翻译区分功能，长文本请求openai, 短文本请求谷歌，微软翻译，则能更好的地优化体验和节约成本。

附上改进的prompt，重点额外一步的润色，代价就是token变得很长：
You are a professional, authentic translation engine, only returns polished translations. 

Please translate the text below which in curly bracket to {{to}} Language step by step
STEP 1. TRANSLATE. Ensure accurate translation.
STEP 2. POLISH. Compare the original text with the result of step 1 and enhance the clarity, grammar, and style to make the translation smoother, easier to understand, and in line with the usage habits of {{to}} while maintaining the original meaning.
REMEMBER: 
- some terms like 'transformer','agent' and 'LLM' should NOT translate.
- ONLY give me the RESULT of STEP 2. NOT Any description .
- When the text is short, it is due to automatic truncation issues. In this case if you feel confused to translate, you can provide the original text directly. 

The text waiting to be translated:
{{text}}

### 提议 | Proposal

在“翻译段落所需的最少字符数”的逻辑上修改应该能简单实现。最后赞扬一下，沉浸式翻译做得非常棒，极大简化了探索外网资源的难度，真棒~

**Comment by theowenyoung** [comment]
实现倒是比较容易，但是就是会引入额外的概念复杂性。

会不会目前的“翻译段落所需的最少字符数” 改成 20 左右，实际上已经能解决99%的问题了？如果有还需要翻译的地方，再配合 “鼠标滑过翻译”，就可以了？

**Reply by jiazhao2048** [reply]
哦哦，还能配合这么用，“最少字符数”不会对“鼠标滑过翻译”生效，试了一下也可以。

那就看在节约的openai,deepl成本是否足够吸引来开发功能了。例如pdf的图例里通常会识别出大量的短文本，也会发起大量的请求。

**Reply by theowenyoung** [reply]
( 我觉得这个有机会在 开发者设置，通过手动编辑配置实现，但是我感觉如果把这个选项加在UI 里，就有点显著增加理解成本了，相当于有一个第一翻译服务，第二翻译服务，再配置一个 字符的分界线..

**Reply by jiazhao2048** [reply]
在开发者设置能够配置也很好~需求是已经比较小众了

如果是UI的话，感觉配个图例如下面的会好理解一些
![image](https://github.com/immersive-translate/immersive-translate/assets/10970061/1cb3c11e-c24a-4a22-8da3-3581bdfaba0e)
